docs/architecture/adr-044-lite-client-with-weak-subjectivity.md

```diff
@@ -0,0 +1,70 @@
+
+This requires providing a user interface for when new nodes connect to the network or when a node that has been offline for longer than the unbending period connects to the network. 
+
+Specifically, the node node needs to initialize the following structure before synching from user input. The expectation is the user will get this information from a trusted source like a validator, https://cosmos.network or other.  A more user friendly solution with trust tradeoff is that we establish an https based protocol with a default end point that populates this information. 
+
+```
+type TrustOptions struct {
+// Required: only trust commits up to this old.
+// Should be equal to the unbonding period.
+TrustPeriod time.Duration `json:"trust-period"`
+
+// Option 1: TrustHeight and TrustHash can both be provided
+// to force the trusting of a particular height and hash.
+// If the latest trusted height/hash is more recent, then this option is
+// ignored.
+TrustHeight int64  `json:"trust-height"`
+TrustHash   []byte `json:"trust-hash"`
+
+// Option 2: Callback can be set to implement a confirmation
+// step if the trust store is uninitialized, or expired.
+Callback func(height int64, hash []byte) error
+}
+```
+
+## Linear Verification
+The linear verification of the light client requires downloading all headers between the `TrustHeight` and the `LatestHeight`. The lite client downloads the full header for the provided `TrustHeight` and then proceeds to download `N+1` headers and applies the [Tendermint validation rules](https://github.com/tendermint/tendermint/blob/master/docs/spec/blockchain/blockchain.md#validation) to each block. 
+
+
+
+## Bisecting Verification
+
+Bisecting Verification is a more bandwidth and compute intensive mechanism that in the most optimistic case requires a light client to only download two block headers to come into synchronization. 
+
+The Bisection algorithm proceeds in the following fashion. The client downloads  and verifies the full block header for `TrustHeight` and then  fetches  `LastestHeight` blocker header.  The client then verifies the `LatestHeight` header.  Finally the client attempts to verify the `LatestHeight` header with voting powers taken from `NextValdiatorSet` in the `TrustHeight` header. This will verification will succeed if that validators from `TrustHeight` still have > 2/3 +1 of voting power in the `LatestHeight`. If this succeeds, the client is fully synchronized. If this fails, then following Bisection Algorithm should be followed.
+
+The Client tries to download the block at the mid-point block between `LatestHeight` and `TrustHeight` and attempts that same algorithm as above using `MidPointHeight` instead of `LatestHeight`. In the case the of failure, recursively perform the `MidPoint` verification until success then start over with an updated `NextValidatorSet` and `TrustHeight`.
```

Comment:
I would note that it's safe to ask the full node to provide the specific headers that the client needs, given the root of trust height that it has. Should we do that instead? The full node has to track a bit more, but further reduces bandwidth / compute for the light client.